lib/buffer.js

```diff
@@ -807,8 +807,8 @@ Buffer.prototype.toJSON = function() {
 
 
 function adjustOffset(offset, length) {
-  offset |= 0;
-  if (offset === 0) {
+  offset = Math.floor(+offset);
+  if (offset === 0 || Number.isNaN(offset)) {
```

Comment:
So, for example, an offset of 3.3 would become 0? That seems like it would be a surprising breaking change.